### ISSUE TITLE
[Enhancement]: improve drag-and-drop example

### ISSUE THREAD

**Issue by makafsal** [carbon-design-system/ibm-products#7644]
### The problem

The current drag-and-drop example [implementation](https://stackblitz.com/github/carbon-design-system/tanstack-carbon/tree/main/web-components/customize-columns?file=src%2Fdnd-example.ts) by @devadula-nandan  lacks accessibility support.

### The solution

From @makafsal 's [investigation findings](https://github.com/carbon-design-system/ibm-products/issues/7528#issuecomment-2934363097) for a perfect dnd library, the team decided to move forward with the native HTML DnD API. So, we've to implement the accessibility support by using native keyboard events.

Hence, extend the current implementation with a11y support.

### Owner/main maintainer(s)

Carbon for IBM Products

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
